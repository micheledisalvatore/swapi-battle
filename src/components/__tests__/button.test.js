import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import dataContext from '../../contexts/data-context'
import Button from '../button'

test('renders the component and clicks on it', () => {
  const play = jest.fn()
  const { container } = render(<dataContext.Provider value={{ play }}><Button /></dataContext.Provider>)

  expect(container).toMatchSnapshot()

  fireEvent.click(screen.getByText('Play'))
  expect(play).toHaveBeenCalled()
})
