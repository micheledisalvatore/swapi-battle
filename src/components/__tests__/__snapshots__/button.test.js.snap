// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component and clicks on it 1`] = `
<div>
  <button
    class="btn btn-primary"
    type="submit"
  >
    Play
  </button>
</div>
`;
